### ISSUE TITLE
URL Encoding for Hyperlinks in PDF

### ISSUE THREAD

**Issue by gregjesl** [doxygen/doxygen#7110]
## Problem
The use of URL-encoded hyperlinks breaks the Latex renderer

It appears that the `%` signs used for URL encoding are treated as comment indicators in Latex.  

## Example
```cpp
/*! \brief A collection of functions pertaining to an <a href="https://en.wikipedia.org/wiki/Irwin%E2%80%93Hall_distribution">Irwin-Hall distribution</a> */
namespace irwinhall
{
  inline float foo() { return 1.0f; }
}
```

Attempting to generate a PDF for the header file listed above will fail.  

## Potential Solution

The Latex generator should escape `%` signs.

**Comment by gregjesl** [comment]
Percentage signs likely need to be escaped here:
https://github.com/doxygen/doxygen/blob/54f758e77fad281c0bcc80c3322cbb1822334b37/src/latexgen.cpp#L1383-L1385

**Comment by albert-github** [comment]
Looks indeed a potential place. It has to be checked what has to be escaped here, or at another (earlier) place and how to escape (probably by means of a backslash before the percent sign).

You didn't mention which version you used, but seen the fact that you mentioned the 54f758e I assume that you checked against the current master version.

**Comment by albert-github** [comment]
I've just pushed a proposed patch, pull request #7111

**Comment by gregjesl** [comment]
> You didn't mention which version you used, but seen the fact that you mentioned the 54f758e I assume that you checked against the current master version.

I was previously using a binary but I just pulled the latest from master and was able to replicate the problem.

**Comment by gregjesl** [comment]
@albert-github #7111 looks good to me!  

> You didn't mention which version you used, but seen the fact that you mentioned the 54f758e I assume that you checked against the current master version 

I was using a compiled binary but I just pulled the latest from master and was able to replicate the issue (I'm glad I did, I like the UI updates I got by pulling the latest :smile:). I also verified your patch works, to include testing the hyperlink.

**Comment by albert-github** [comment]
Code has been integrated in master on github (please don't close the issue as this will be done at the moment of a release).

**Comment by doxygen** [comment]
This issue was previously marked 'fixed but not released',
which means it should be fixed in doxygen version 1.8.16.
Please verify if this is indeed the case. Reopen the
issue if you think it is not fixed and please include any additional information
that you think can be relevant (preferably in the form of a self-contained example).
